lib/graphql/graphql-provider.ts

```diff
@@ -0,0 +1,61 @@
+import { ChainId } from '@uniswap/sdk-core'
+
+import { GraphQLClient, IGraphQLClient } from './graphql-client'
+import {
+  GRAPHQL_QUERY_MULTIPLE_TOKEN_INFO_BY_CONTRACTS,
+  GRAPHQL_QUERY_TOKEN_INFO_BY_ADDRESS_CHAIN,
+} from './graphql-queries'
+import { TokenInfoResponse, TokensInfoResponse } from './graphql-schemas'
+
+/* Interface for accessing Uniswap GraphQL API */
+export interface IUniGraphQLProvider {
+  /* Fetch token info for a given chain and address */
+  getTokenInfo(chainId: ChainId, address: string): Promise<TokenInfoResponse>
+  /* Fetch token info for multiple tokens given a chain and addresses */
+  getTokensInfo(chainId: ChainId, addresses: string[]): Promise<TokensInfoResponse>
+  // Add more methods here as needed.
+  // - more details: https://github.com/Uniswap/data-api-graphql/blob/main/graphql/schema.graphql
+}
+
+/* Implementation of the UniGraphQLProvider interface to give access to Uniswap GraphQL API */
+export class UniGraphQLProvider implements IUniGraphQLProvider {
+  private readonly endpoint = process.env.UNI_GRAPHQL_ENDPOINT!
+  private readonly headers = {
+    Origin: process.env.UNI_GRAPHQL_HEADER_ORIGIN!,
+    'Content-Type': 'application/json',
+  }
```

Comment:
We probably need to add these env variables to github actions for the tests to pass, but we should also have landed on the fallback for those tests that are failing?

Reply:
initialization is outside our try/catch which makes sense, we want to fail here and catch this in tests.
will add envs to gh actions

Reply:
added - please tal that looks ok
tested against persoanl aws as well